test/nuts/agent.nut.ts

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { join } from 'node:path';
+import { readdirSync, readFileSync, writeFileSync } from 'node:fs';
+import { expect } from 'chai';
+import { genUniqueString, TestSession } from '@salesforce/cli-plugins-testkit';
+import { Connection, Org, SfProject, User, UserFields } from '@salesforce/core';
+import { ComponentSetBuilder } from '@salesforce/source-deploy-retrieve';
+import { sleep } from '@salesforce/kit';
+import { Agent, type AgentJobSpec, type AgentJobSpecCreateConfig } from '../../src/index';
+
+/* eslint-disable no-console */
+
+describe('agent NUTs', () => {
+  const agentName = 'The Campus Agent Test';
+  let session: TestSession;
+  let connection: Connection;
+  let defaultOrg: Org;
+  let agentSpec: AgentJobSpec;
+
+  before(async () => {
+    session = await TestSession.create({
+      project: {
+        sourceDir: join('test', 'projects', 'agent-project'),
+      },
+      devhubAuthStrategy: 'AUTO',
+      scratchOrgs: [
+        {
+          setDefault: true,
+          config: join('config', 'project-scratch-def.json'),
+        },
+      ],
+    });
+    const username = session.orgs.get('default')!.username as string;
+    defaultOrg = await Org.create({ aliasOrUsername: username });
+    connection = defaultOrg.getConnection();
+
+    // assign the EinsteinGPTPromptTemplateManager to the scratch org admin user
+    const queryResult = await connection.singleRecordQuery<{ Id: string }>(
+      `SELECT Id FROM User WHERE Username='${username}'`
+    );
+    const user = await User.create({ org: defaultOrg });
+    await user.assignPermissionSets(queryResult.Id, ['EinsteinGPTPromptTemplateManager']);
+  });
+
+  after(async () => {
+    await session?.clean();
+  });
+
+  // skipping because the server API is not reliable enough to run in CI
+  it.skip('should create an agent spec', async () => {
```

Comment:
is there a WI to track when these should be re-enabled?

Reply:
No.  I can create one.  I was planning to try the tests after R2.